pallets/rmrk-core/src/tests.rs

```diff
@@ -1814,32 +1821,116 @@ fn set_property_works() {
+		assert_eq!(
+			RMRKCore::properties((0, Some(0), key.clone())).unwrap(),
+			value_immutable.clone()
+		);
 		// BOB does not own NFT so attempt to set property should fail
 		assert_noop!(
-			RMRKCore::set_property(Origin::signed(BOB), 0, Some(0), key.clone(), value.clone()),
+			RMRKCore::set_property(
+				Origin::signed(BOB),
+				0,
+				Some(0),
+				key.clone(),
+				value_immutable.clone()
+			),
 			Error::<Test>::NoPermission
 		);
 	});
 }
 
+#[test]
+fn set_mutable_property_works() {
+	ExtBuilder::build().execute_with(|| {
+		// Define property key
+		let key = stbk("test-key");
+		// Define property value (immutable because collection issue should skip the check)
+		let value_immutable = PropertyValue { mutable: false, value: stb("test-value0") };
+		let value_mutable = PropertyValue { mutable: true, value: stb("test-value1") };
+
+		// Create a basic collection
+		assert_ok!(basic_collection());
+		// Mint NFT
+		assert_ok!(basic_mint(0));
+		// ALICE sets property on NFT
+		assert_ok!(RMRKCore::set_property(
+			Origin::signed(ALICE),
+			0,
+			Some(0),
+			key.clone(),
+			value_immutable.clone()
+		));
+		// Successful property setting should trigger a PropertySet event
+		System::assert_last_event(MockEvent::RmrkCore(crate::Event::PropertySet {
+			collection_id: 0,
+			maybe_nft_id: Some(0),
+			key: key.clone(),
+			value: value_immutable.clone(),
+		}));
+		// Property value now exists
+		assert_eq!(
+			RMRKCore::properties((0, Some(0), key.clone())).unwrap(),
+			value_immutable.clone()
+		);
+		// ALICE sets property on NFT
+		assert_ok!(RMRKCore::set_property(
+			Origin::signed(ALICE),
+			0,
+			Some(0),
+			key.clone(),
+			value_mutable.clone()
+		));
+		// Successful property setting should trigger a PropertySet event
+		System::assert_last_event(MockEvent::RmrkCore(crate::Event::PropertySet {
+			collection_id: 0,
+			maybe_nft_id: Some(0),
+			key: key.clone(),
+			value: value_mutable.clone(),
+		}));
+		// Property value now exists
+		assert_eq!(RMRKCore::properties((0, Some(0), key.clone())).unwrap(), value_mutable.clone());
+		// ALICE sends NFT to BOB
+		assert_ok!(RMRKCore::nft_send(ALICE, 0, 0, AccountIdOrCollectionNftTuple::AccountId(BOB)));
+		// BOB owns the NFT, so he should be able to change the value, but not the mutability
+		assert_ok!(RMRKCore::set_property(
+			Origin::signed(BOB),
+			0,
+			Some(0),
+			key.clone(),
+			value_immutable.clone()
+		));
+		// Successful property setting should trigger a PropertySet event
+		System::assert_last_event(MockEvent::RmrkCore(crate::Event::PropertySet {
+			collection_id: 0,
+			maybe_nft_id: Some(0),
+			key: key.clone(),
+			value: PropertyValue { mutable: true, value: value_immutable.value.clone() },
+		}));
+		// Property value now be changed
+		assert_eq!(
+			RMRKCore::properties((0, Some(0), key.clone())).unwrap(),
+			PropertyValue { mutable: true, value: value_immutable.value.clone() }
+		);
+	});
+}
+
```

Comment:
Would be good to add extra test for error handling ( `NoPermission` ). Ie when BOB tries to mutate immutable prop.

Reply:
Added the test case.